opensearch_py_ml/ml_commons/rag_pipeline/rag/AIConnectorHelper.py

```diff
@@ -0,0 +1,496 @@
+# SPDX-License-Identifier: Apache-2.0
+# The OpenSearch Contributors require contributions made to
+# this file be licensed under the Apache-2.0 license or a
+# compatible open source license.
+# Any modifications Copyright OpenSearch Contributors. See
+# GitHub history for details.
+
+import boto3
+import json
+import requests
+from requests.auth import HTTPBasicAuth
+from requests_aws4auth import AWS4Auth
+import time
+from opensearchpy import OpenSearch, RequestsHttpConnection
+from urllib.parse import urlparse
+
+from opensearch_py_ml.ml_commons.rag_pipeline.rag.IAMRoleHelper import IAMRoleHelper
+from opensearch_py_ml.ml_commons.rag_pipeline.rag.SecretsHelper import SecretHelper
```

Comment:
These two can be used by other features. And this shouldn't be part of rag_pipeline only. Can we put this outside of the rag_pipeline directory?